post_install.py fails if GTK 3 is not installed (granite-7)

### What Happened?

Looking forward to seeing the 7.0 release! Unfortunately the build currently fails on NixOS with `No such file or directory: 'gtk-update-icon-cache'`:

<details>

```
Running custom install script '/build/source/meson/post_install.py --iconsdir /nix/store/y68x4jpih18vd3vk6j6fgivv8xiq1fpb-granite/share/icons/hicolor'
--- stdout ---
Compiling icon cache ...

--- stderr ---
Traceback (most recent call last):
  File "/build/source/meson/post_install.py", line 15, in <module>
    subprocess.run(['gtk-update-icon-cache', icons_dir])
  File "/nix/store/i6vabb4div9iy6lsl642d86k1q8riasn-python3-3.9.9/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/nix/store/i6vabb4div9iy6lsl642d86k1q8riasn-python3-3.9.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/nix/store/i6vabb4div9iy6lsl642d86k1q8riasn-python3-3.9.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'gtk-update-icon-cache'

FAILED: install script '/build/source/meson/post_install.py --iconsdir /nix/store/y68x4jpih18vd3vk6j6fgivv8xiq1fpb-granite/share/icons/hicolor' exit code 1, stopped
FAILED: meson-install 
/nix/store/gbplxvpxh13gfm9ppdg1qlhxp9kasrn6-meson-0.60.3/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

</details>

A fix for this is change `gtk-update-icon-cache` to `gtk4-update-icon-cache` in `post_install.py`, or use [`gnome.post_install()`](https://github.com/mesonbuild/meson/blob/master/docs/markdown/Gnome-module.md#gnomepost_install) (requires Meson 0.57.0 or later, and [an example for this](https://gitlab.gnome.org/GNOME/gnome-text-editor/-/commit/0a9bca3e6460435ca8a2c96e4a6af7ef920465d4)), but for granite after https://github.com/elementary/granite/pull/548 maybe we can just drop the post_install script. 

https://github.com/elementary/granite/blob/c14a5915d45e362cd550d08d80096ffc85ef4572/meson/post_install.py#L15

This will probably become an issue for every Pantheon package that is ported to GTK 4 and does `gtk-update-icon-cache` in NixOS so many thanks for reading this!

### Steps to Reproduce

Build granite-7 (e8e9f51c6dab1f7ddb827e3bacf6cd5ed6ad7e08) on NixOS, remove `gtk3` from dependencies.

### Expected Behavior

Package builds.

### OS Version

Other Linux

### Software Version

Compiled from git

### Log Output

_No response_

### Hardware Info

_No response_

--- a/meson/post_install.py
+++ b/meson/post_install.py
@@ -12,5 +12,5 @@ icons_dir = args["iconsdir"]
 
 if not os.environ.get('DESTDIR'):
     print('Compiling icon cache ...')
-    subprocess.run(['gtk-update-icon-cache', icons_dir])
+    subprocess.run(['gtk4-update-icon-cache', icons_dir])
 

